opslevel/datasource_opslevel_domains_all.go

```diff
@@ -26,20 +23,20 @@ type DomainDataSourcesAll struct {
 	CommonDataSourceClient
 }
 
-// domainObjectType is derived from DomainDataSourceModel, needed for lists
-var domainObjectType = types.ObjectType{
-	AttrTypes: map[string]attr.Type{
-		"aliases":     types.ListType{ElemType: types.StringType},
-		"description": types.StringType,
-		"id":          types.StringType,
-		"name":        types.StringType,
-		"owner":       types.StringType,
-	},
+// DomainDataSourcesAllModel describes the data source data model.
+type DomainDataSourcesAllModel struct {
+	Domains []domainDataSourceModel `tfsdk:"domains"`
 }
 
-// DomainDataSourceModel describes the data source data model.
-type DomainDataSourcesModel struct {
-	Domains types.List `tfsdk:"domains"`
+func NewDomainDataSourcesAllModel(ctx context.Context, domains []opslevel.Domain) (DomainDataSourcesAllModel, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	domainModels := []domainDataSourceModel{}
+	for _, domain := range domains {
+		domainModel, domainDiag := NewDomainDataSourceModel(ctx, domain)
+		diags = append(diags, domainDiag...)
```

Comment:
```suggestion
		if diags == nil {
		    diags = domainDiag
		} else {
		    diags.Append(domainDiag...)
		}
```

Would it be better to use the `.Append()` here? 

https://developer.hashicorp.com/terraform/plugin/framework/diagnostics#working-with-existing-diagnostics